backend/report/views.py

```diff
@@ -0,0 +1,299 @@
+                    daily_avg_single_annotation_time.append(single_annotation_time_per_day)
+                for time in daily_avg_single_annotation_time:
+                    progressed_users_avg_annotation_time_per_day.append(time)
+        if progressed_users_avg_annotation_time_per_day:
+            df = pd.DataFrame(progressed_users_avg_annotation_time_per_day)
+            df = df.groupby('date')['average_annotation_time (seconds)'].mean().reset_index(name='avg_annotation_time_single_text (seconds)')
+            all_user_daily_avg_single_annotation_time = df.to_dict(orient='records')
+            for time in all_single_annotation_time:
+                progressed_users_avg_annotation_time.append(time)
+            all_user_average_annotation_time = sum(progressed_users_avg_annotation_time) / len(progressed_users_avg_annotation_time)
+            data = {"average_annotation_time (seconds)": all_user_average_annotation_time, "daily_average_annotation_time": all_user_daily_avg_single_annotation_time}
+            return Response(data=data, status=status.HTTP_200_OK)
+        else:
+            return Response(data={"average_annotation_time (seconds)": 0, "daily_average_annotation_time": []}, status=status.HTTP_200_OK)
+
+
+class AllUsersAvgDailyAnnotationTimeAPI(APIView):
+    permission_classes = [IsAuthenticated & (IsProjectAdmin | IsProjectStaffAndReadOnly)]     
+
+    def get(self, request, *args, **kwargs):
+        all_user_ids = User.objects.all().values_list('id', flat=True)
+        startdate = self.kwargs["startdate"]
+        enddate = self.kwargs["enddate"]
+        progressed_users_annotation_time_per_day = []
+        for user_id in all_user_ids:
+            confirmed_examples_by_user = ExampleState.objects.get_confirmed_time_by_example([user_id], startdate, enddate)
+            if len(confirmed_examples_by_user) != 0:
+                confirmed_df = pd.DataFrame(confirmed_examples_by_user)
+                confirmed_df['Date'] = confirmed_df['confirmed_at'].dt.strftime('%Y-%m-%d')
+                confirmed_dates = list(confirmed_df.Date.unique())
+                daily_user_annotation_time = []
+                all_annotation_time = []
+                for date in confirmed_dates:
+                    requested_examples = confirmed_df[confirmed_df['Date'] == date].example_id
+                    daily_single_annotation_time = []
+                    for example_id in requested_examples:
+                        started_examples_by_user = ExampleAnnotateStartState.objects.get_started_time_by_example(example_id, [user_id], startdate, enddate)
+                        annotation_time  = confirmed_df[confirmed_df['example_id']==example_id]["confirmed_at"] - started_examples_by_user[0]['started_at']
+                        daily_single_annotation_time.append(annotation_time.dt.seconds.values[0])
+                        all_annotation_time.append(annotation_time.dt.seconds.values[0])
```

Comment:
Similar case, I got an index out of range error so I added `if len(started_examples_by_user) > 0:` in my local environment and put line 236 - 238 under the if block

Reply:
Similar case here also :sweat_smile: